pyccel/parser/syntax/omp_versions/version_4_5.py

```diff
@@ -0,0 +1,585 @@
+# pylint: disable=protected-access, missing-function-docstring
+"""
+Module containing the grammar rules for the OpenMP 4.5 specification,
+"""
+from inspect import isclass
+import ast as Ast
+import pyccel.ast.omp as PreviousVersion
+from pyccel.ast.core import FunctionCall, Return
+from pyccel.ast.omp import (
+    OmpConstruct,
+    OmpEndConstruct,
+)
+from pyccel.ast.variable import Variable
+from pyccel.parser.extend_tree import extend_tree
+from pyccel.ast.datatypes import NativeInteger, NativeVoid
+
+__all__ = ("Openmp",)
+
+
+class Openmp:
```

Comment:
Am I right that this class is behaving as a namespace?

Reply:
Yes, it is. it's just a class that groups classes from the same version together and also help new version to inherit the same classes of the previous version

so for example version_5_0 can just have class Openmp(version_4_5.Openmp) and this will make version_5_0 Openmp class fully ready to be used by Textx and if we can only override the classes that got changes in version_5_1